### DISCUSSION TITLE
Approximate ETA of MVP?

### DISCUSSION THREAD

**Discussion by josephrocca** [gpuweb/gpuweb#2003] [Q&A] [answered]
I know it's notoriously difficult to make estimates on large multi-stakeholder projects like this, but as a dev who is super interested in using this, and who doesn't have the technical knowhow to make my own rough estimates based on [the remaining tasks in the MVP milestone](https://github.com/gpuweb/gpuweb/milestone/2) and the [Main/WGSL project boards](https://github.com/orgs/gpuweb/projects), I'm interested in whether any of the pros here could hazard even a very low-resolution guess at when the MVP of WebGPU + WGSL might hit browsers (unflagged)?

By low-resolution, I mean something like "maybe within the next N to N+1 years". Or are we still very firmly in the "it'll be ready when it's ready" stage? Obviously I'd love a high-res (Q**X** 202**Y**) estimate if someone is that confident, but I'd be very happy even with a super low-res guess.

Thanks! 🙏

**Comment by Kangz** [comment, marked as answer]
The group previously mentioned they wanted to target end of 2021 for standardization, but that deadline has to be fluid given we are a standards body. There's great progress being done though, so maybe it can be done.

In terms of implementation, on the Chromium side we are looking to do an [Origin Trial](https://developer.chrome.com/origintrials/)  for WebGPU starting with [Chromium 94](https://chromiumdash.appspot.com/schedule). Think of it as an open beta so we can get feedback from developers like you! Firefox's implementation is a bit behind, but mostly interoperable already (except when we implement spec changes at a different rate). WebKit just deleted the remnant of their WebGPU implementation because it was severely out of date and just increasing binary size / compile times. [However in the relevant issue they noted that there is still strong desire to have WebGPU in WebKit](https://bugs.webkit.org/show_bug.cgi?id=228179#c21).

So most like within one year, maybe within two if we hit some unknown roadblock. The exact time will depend on the browsers obvisouly.

**Reply by josephrocca** [reply]
Thank you for the excellent summary!
